CMakeLists.txt

```diff
@@ -1,84 +1,58 @@
 # SPDX-FileCopyrightText: 2008-2021 HPDCS Group <rootsim@googlegroups.com>
 # SPDX-License-Identifier: GPL-3.0-only
 cmake_minimum_required(VERSION 3.10)
-project("ROOT-Sim core" LANGUAGES C)
-set(PROJECT_VERSION 3.0.0-alpha.3)
-set(PROJECT_DESCRIPTION "A General-Purpose Multithreaded Parallel/Distributed Simulation Library")
+project("ROOT-Sim core" LANGUAGES C DESCRIPTION "A General-Purpose Multi-threaded Parallel/Distributed Simulation Library")
 
-#set(CMAKE_VERBOSE_MAKEFILE ON)
-
-if (MSVC)
-    # warning level 4 and all warnings as errors
-    add_compile_options(/Wall)
-else()
-    # lots of warnings and all warnings as errors
-    add_compile_options(-Wall -Wextra -pedantic)
-endif()
-
-add_definitions(-DROOTSIM_VERSION="${PROJECT_VERSION}")
-
-set(srcs
-        src/arch/io.c
-        src/arch/mem.c
-        src/arch/thread.c
-        src/core/core.c
-        src/init.c
-        src/core/sync.c
-        src/datatypes/msg_queue.c
-        src/distributed/control_msg.c
-        src/distributed/mpi.c
-        src/gvt/fossil.c
-        src/gvt/gvt.c
-        src/gvt/termination.c
-        src/lib/lib.c
-        src/lib/random/random.c
-        src/lib/state/state.c
-        src/lib/topology/topology.c
-        src/log/file.c
-        src/log/log.c
-        src/log/stats.c
-        src/lp/lp.c
-        src/lp/process.c
-        src/mm/auto_ckpt.c
-        src/mm/buddy/buddy.c
-        src/mm/buddy/ckpt.c
-        src/mm/msg_allocator.c
-        src/parallel/parallel.c
-        src/serial/serial.c)
+set(PROJECT_VERSION 3.0.0-alpha.3) # todo: use cmake project VERSION argument when we are out of the alpha
```

Comment:
I think that, since CMake provides a proper VERSION argument, we should use it, if possible. The problem that you underline with the ```bump_version.py``` script can be solved with a simple regex. That said, I have no strong preference on this point.